assets/src/edit-story/components/panels/elementAlignment.js

```diff
@@ -0,0 +1,333 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as AlignBottom } from '../../icons/align_bottom.svg';
+import { ReactComponent as AlignTop } from '../../icons/align_top.svg';
+import { ReactComponent as AlignCenter } from '../../icons/align_center.svg';
+import { ReactComponent as AlignMiddle } from '../../icons/align_middle.svg';
+import { ReactComponent as AlignLeft } from '../../icons/align_left.svg';
+import { ReactComponent as AlignRight } from '../../icons/align_right.svg';
+import { ReactComponent as HorizontalDistribute } from '../../icons/horizontal_distribute.svg';
+import { ReactComponent as VerticalDistribute } from '../../icons/vertical_distribute.svg';
+import { dataPixels } from '../../units/dimensions';
+import getCommonValue from './utils/getCommonValue';
+import getBoundRect, {
+  calcRotatedObjectPositionAndSize,
+} from './utils/getBoundRect';
+
+const ElementRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  padding: 10px 20px;
+  border-top: 1px solid ${({ theme }) => theme.colors.bg.v9};
+`;
+
+const IconButton = styled.button`
+  display: flex;
+  width: 28px;
+  height: 28px;
+  justify-content: center;
+  align-items: center;
+  background-color: unset;
+  cursor: pointer;
+  padding: 0;
+  border: 0;
+  border-radius: 4px;
+
+  &:hover {
+    background-color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.1)};
+  }
+
+  ${({ disabled }) =>
+    disabled &&
+    `
+    pointer-events: none;
+		opacity: .2;
+	`}
+
+  svg {
+    color: ${({ theme }) => theme.colors.mg.v2};
+    width: 28px;
+    height: 28px;
+  }
+`;
+
+const SeparateBorder = styled.div`
+  border-left: 1px dashed ${({ theme }) => rgba(theme.colors.bg.v0, 0.3)};
+  height: 12px;
+  margin-left: 4px;
+  margin-right: 4px;
+`;
+
+function ElementAlignmentPanel({ selectedElements, onSetProperties }) {
+  const boundRect = getBoundRect(selectedElements);
+  const isFill = getCommonValue(selectedElements, 'isFill');
+
+  const isJustifyEnabled = isFill || selectedElements.length < 2;
+  const isDistributionEnabled = isFill || selectedElements.length < 3;
+
+  const handleAlignLeft = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetX = 0;
+      if (rotationAngle) {
+        const { width: newWidth } = calcRotatedObjectPositionAndSize(
```

Comment:
Nit: let's name it something like "outer width" or "frame width". Otherwise it took me couple of min to figure out why would width change for alignment :)

Reply:
Updated to frameWidth! Sorry for the confusion. ;)